[WIP] Road to version 9.0: update to go 1.20 and operator-sdk 1.33

brokenpip3 opened the pull request.

ansh-devs commented:
Hi @brokenpip3 , Can I pick this one up ?

brokenpip3 commented:
@ansh-devs hi :wave: 

thanks for stepping in :tada: since I don't have time for this in the following weeks.
I fixed all the conflict, merged master and starting fixing the golang-ci linter warnings and some of code problems after the golang and library update.

Feel free to open another branch and work on top of this one :)

My suggestion is to start from fixing the build and then the ci linter warnings. 
Let me know if you want to work on this in the following days, in case we can also have a chat in gitter if you have any doubts :)

ansh-devs commented:
> @ansh-devs hi 👋
> 
> thanks for stepping in 🎉 since I don't have time for this in the following weeks. I fixed all the conflict, merged master and starting fixing the golang-ci linter warnings and some of code problems after the golang and library update.
> 
> Feel free to open another branch and work on top of this one :)
> 
> My suggestion is to start from fixing the build and then the ci linter warnings. Let me know if you want to work on this in the following days, in case we can also have a chat in gitter if you have any doubts :)

@brokenpip3 Ok, I will continue with the refactoring taking this as the base 🛠️

ansh-devs commented:
@ansh-devs/kubernetes-operaror@b7f4e25b71368408d3d1702e34b0b9238bd71660

ansh-devs commented:
@brokenpip3  
https://github.com/jenkinsci/kubernetes-operator/blob/f6d801810670194b4b8a88224e092ca2f6f3f8ee/cmd/main.go#L132

is this port denotes the webhook server port ?

ansh-devs commented:
Hi @brokenpip3, I have fixed the breaking changes also updated to go version to 1.22.
- [ ] Next step will be fixed the E2E and HELM test

ansh-devs commented:
@ansh-devs/kubernetes-operaror@b7f4e25b71368408d3d1702e34b0b9238bd71660

@brokenpip3 how can I join the Slack workspace

brokenpip3 commented:
> Hi @brokenpip3, I have fixed the breaking changes also updated to go version to 1.22.
> ### Changes -


Great great progress! :clap: 
Thanks for taking care of this part :)

I saw that you are working on your fork, could you please open a PR against this branch? You can just open a normal PR but instead of targeting `master` you can target this branch: `version-9.0`.
I'm asking because for me will be much easier to help you in case of questions and to start reviewing the code :)

>@brokenpip3 how can I join the Slack workspace

There is no slack workplace, you can join the matrix room from any client you want, gitter for instance.
But I saw you already joned it so we can continue the discussion there if you like :)
Or you can write me an email here `<mygithubusername>[at]gmail[dot].com`

ansh-devs commented:
@brokenpip3  Refer this - #1037

brokenpip3 commented:
superseded by https://github.com/jenkinsci/kubernetes-operator/pull/1094
